src/Facets/RelayFacet.sol

```diff
@@ -0,0 +1,221 @@
+            revert InvalidQuote();
+        }
+        _;
+    }
+
+    /// Constructor ///
+
+    constructor(address _relayReceiver, address _relaySolver) {
+        relayReceiver = _relayReceiver;
+        relaySolver = _relaySolver;
+    }
+
+    /// External Methods ///
+
+    /// @notice Bridges tokens via Relay
+    /// @param _bridgeData The core information needed for bridging
+    /// @param _relayData Data specific to Relay
+    function startBridgeTokensViaRelay(
+        ILiFi.BridgeData calldata _bridgeData,
+        RelayData calldata _relayData
+    )
+        external
+        payable
+        nonReentrant
+        onlyValidQuote(_bridgeData, _relayData)
+        refundExcessNative(payable(msg.sender))
+        validateBridgeData(_bridgeData)
+        doesNotContainSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
+    {
+        LibAsset.depositAsset(
+            _bridgeData.sendingAssetId,
+            _bridgeData.minAmount
+        );
+        _startBridge(_bridgeData, _relayData);
+    }
+
+    /// @notice Performs a swap before bridging via Relay
+    /// @param _bridgeData The core information needed for bridging
+    /// @param _swapData An array of swap related data for performing swaps before bridging
+    /// @param _relayData Data specific to Relay
+    function swapAndStartBridgeTokensViaRelay(
+        ILiFi.BridgeData memory _bridgeData,
+        LibSwap.SwapData[] calldata _swapData,
+        RelayData calldata _relayData
+    )
+        external
+        payable
+        nonReentrant
+        onlyValidQuote(_bridgeData, _relayData)
+        refundExcessNative(payable(msg.sender))
+        containsSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
+        validateBridgeData(_bridgeData)
+    {
+        _bridgeData.minAmount = _depositAndSwap(
+            _bridgeData.transactionId,
+            _bridgeData.minAmount,
+            _swapData,
+            payable(msg.sender)
+        );
+        _startBridge(_bridgeData, _relayData);
+    }
+
+    /// Internal Methods ///
+
+    /// @dev Contains the business logic for the bridge via Relay
+    /// @param _bridgeData The core information needed for bridging
+    /// @param _relayData Data specific to Relay
+    function _startBridge(
+        ILiFi.BridgeData memory _bridgeData,
+        RelayData calldata _relayData
+    ) internal {
+        // check if sendingAsset is native or ERC20
+        if (LibAsset.isNativeAsset(_bridgeData.sendingAssetId)) {
+            // Native
+
+            // Send Native to relayReceiver along with requestId as extra data
+            (bool success, bytes memory reason) = relayReceiver.call{
+                value: _bridgeData.minAmount
+            }(abi.encode(_relayData.requestId));
+            if (!success) {
+                revert(LibUtil.getRevertMsg(reason));
+            }
+        } else {
+            bytes memory quoteId = _relayData.callData[68:];
```

Comment:
Is this `quoteId` something else than the `requestId` we use for nativ token transfers?

Reply:
The quoteid is whatever is concatenated to the end of the calldata. Sometimes this is the requestid and sometimes its something different.

Reply:
@ezynda3 it we are not using the `requestId` our validation does not work, the `quoteId` could contain something else that what we validated. Please check with the relay team if we can enfoce to only use the `requestId` here and in their generated calldata.